### ISSUE TITLE
Addon-docs: Relative links are broken in descriptions

### ISSUE THREAD

**Issue by shilman** [storybookjs/storybook#14360]
Extracted from previous issue because this repro is a special case.

===

Hi @shilman,

I've just created a repo: https://github.com/mgameover/sb-md-relative-links Please take a look.

**To reproduce the issue:**
1. npm run storybook
2. goto "Markdowm / Link test" story, "Docs" tab
3. click on "test link"

**Result:**
You will be navigated to http://localhost:6006/iframe.html?path=/docs/markdown--target
**Expected result:**
You should be navigated to "Markdown / Target" story, "Docs" tab

Hope that helps

_Originally posted by @mgameover in https://github.com/storybookjs/storybook/issues/13630#issuecomment-808423309_

**Comment by mgameover** [comment]
Hi guys, @shilman,
Have you got a chance to take a look at that?

P.S. I updated the repo with storybook to v6.2.9, the issue is still there.
Thanks!

**Comment by thien-do** [comment]
Tried with 6.3.0 (beta), the issue is still there :D

**Comment by anatel** [comment]
I have the same problem :(

**Comment by mgameover** [comment]
I've just tried [v6.3.3](https://github.com/mgameover/sb-md-relative-links/tree/main) and [v6.4.0-alpha.12](https://github.com/mgameover/sb-md-relative-links/tree/feature/sb-6.4.0-alpha.12), the issue is still there.
Are there any plans to fix that?

**Comment by zolk** [comment]
I just started a new client engagement and this issue has come back to haunt me after causing problems at my last client. 😢 Appears to be still broken in `6.4.0-alpha.34`.

**Comment by shilman** [comment]
@zolk (and others) if you want this fixed, please upvote by adding a 👍 to the issue description. We use this to help prioritize! thanks! 🙏

**Comment by mgameover** [comment]
@shilman, yes, this is highly anticipated and we are still waiting for the fix. 

I had to use **monkey patching** approach by patching `Link` function in `\node_modules\@storybook\components\dist\esm\typography\link\DocumentFormatting.js`. 
Definitely, that's not a right way. But we couldn't stay with older version of Storybook due to angular 12 migration.

**Comment by vnys** [comment]
@mgameover Can you share the code for that, please? I have the same problem…

**Comment by mgameover** [comment]
Hi @vnys, 
I added [patch.js](https://github.com/mgameover/sb-md-relative-links/blob/main/tasks/patch.js) to the sample [repo](https://github.com/mgameover/sb-md-relative-links).
Once storybook has been installed, you can run the patch with the following command: `npm run patch-storybook`.

**Warning**: That's definitely not a right way of fixing this issue, so use it at your own risk.

**Comment by mgameover** [comment]
Hey @shilman, 
I create PR, please take a look.
Thanks.

**Comment by adamdiestelkamp** [comment]
Thanks @stefanhamburger for that, was having this issue and that resolved it!

**Comment by svetlovaalina** [comment]
> As a workaround, we were able to use a relative link like `../?path=/docs/lorem-ipsum`; this worked for both links in MDX stories, as well as links in components' JSDoc comments. Hope this helps until this issue is resolved.

Thanks a lot, it helped me too
